sql/src/main/java/io/crate/operation/scalar/cast/AbstractCastFunction.java

```diff
@@ -94,8 +95,11 @@ protected Symbol onNormalizeException(Symbol argument, Throwable t) {
     }
 
     protected To onEvaluateException(From argument, Throwable t) {
-        Throwables.propagate(t);
-        return null;
+        String value = argument instanceof BytesRef ? ((BytesRef)argument).utf8ToString() : argument.toString();
+        throw new IllegalArgumentException(
+                String.format(Locale.ENGLISH, "cannot cast '%s' to type %s",
```

Comment:
the error message is now different to the error message that happens in normalizeSymbol (onNormalizeException). And there is also a special ConversionException which is raised from `Literal.convert` which is also used to do casts (if all arguments are literals)